plonky_block_proof_gen/src/verifier_state.rs

```diff
@@ -0,0 +1,110 @@
+//! This module defines the `VerifierState`, that contains the necessary data to
+//! handle succinct block proofs verification.
+
+use std::ops::Range;
+
+use log::info;
+use paste::paste;
+use plonky2_evm::{all_stark::AllStark, config::StarkConfig};
+
+use crate::constants::*;
+use crate::{
+    prover_state::ProverState,
+    types::{AllRecursiveCircuits, VerifierData},
+};
+
+/// Plonky2 verifier state.
+///
+/// The default generation requires generating all the verifier data before
+/// extracting the verifier-related data, which can take a long time and require
+/// a large amount of memory.
+pub struct VerifierState {
+    ///
+    pub state: VerifierData,
+}
+
+/// Builder for the verifier state.
+#[derive(Debug)]
+pub struct VerifierStateBuilder {
+    arithmetic_circuit_size: Range<usize>,
+    byte_packing_circuit_size: Range<usize>,
+    cpu_circuit_size: Range<usize>,
+    keccak_circuit_size: Range<usize>,
+    keccak_sponge_circuit_size: Range<usize>,
+    logic_circuit_size: Range<usize>,
+    memory_circuit_size: Range<usize>,
+}
+
+impl Default for VerifierStateBuilder {
+    fn default() -> Self {
+        // The default ranges are somewhat arbitrary, but should be enough for testing
+        // purposes against most transactions.
+        // Some heavy contract deployments may require bumping these ranges though.
+        Self {
+            arithmetic_circuit_size: DEFAULT_ARITHMETIC_RANGE,
+            byte_packing_circuit_size: DEFAULT_BYTE_PACKING_RANGE,
+            cpu_circuit_size: DEFAULT_CPU_RANGE,
+            keccak_circuit_size: DEFAULT_KECCAK_RANGE,
+            keccak_sponge_circuit_size: DEFAULT_KECCAK_SPONGE_RANGE,
+            logic_circuit_size: DEFAULT_LOGIC_RANGE,
+            memory_circuit_size: DEFAULT_MEMORY_RANGE,
+        }
+    }
+}
+
+macro_rules! define_set_circuit_size_method {
```

Comment:
While I understand why you added this I think it's wort pondering whether we can get rid of the need to set these values completely. It is good to keep the verifier as "dumb" as possible.

Reply:
I think I'm missing a crucial piece here. Is the verifier stateful in the sense that it needs data from previous proofs to verify the next one?

Reply:
Did our offline discussion clear the confusion?